xrootd/client/session.go

```diff
@@ -163,88 +162,91 @@ func (sess *session) Close() error {
 	return nil
 }
 
+// handleReadError handles an error encountered while reading and parsing a response.
+// If the current session is equal to the initial, the error is considered critical and handleReadError panics.
+// Otherwise, the current session is closed and all requests are redirected to the initial session.
+// See http://xrootd.org/doc/dev45/XRdv310.pdf, p. 11 for details.
+func (sess *session) handleReadError(err error) {
+	if sess.sessionID == sess.client.initialSessionID {
+		// TODO: what should we do in case initial session is aborted?
+		// Should we try to reconnect to the server and re-issue all requests?
+		panic(err)
+	}
+	sess.mu.RLock()
+	resp := mux.ServerResponse{Redirection: &mux.Redirection{Addr: sess.client.initialSessionID}}
+	for streamID := range sess.requests {
+		err := sess.mux.SendData(streamID, resp)
+		// TODO: should we log error somehow? We have nowhere to send it.
+		_ = err
+	}
+	sess.mu.RUnlock()
+	sess.Close()
+}
+
+// handleWaitResponse handles a "kXR_wait" response by re-issuing the request with streamID
+// after the number of seconds encoded in data.
+// See http://xrootd.org/doc/dev45/XRdv310.pdf, p. 35 for the specification of the response.
+func (sess *session) handleWaitResponse(streamID xrdproto.StreamID, data []byte) error {
```

Comment:
according to the coverage report, it seems `handleWaitResponse` isn't exercized in our tests.
could we fix this?